### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by nervo**
Among others:
- Tests everywhere \o/
- Generalize lookup plugins
- Better documentation
- Handle extension enabling/disabling
- Treats sapi/extensions package dependencies
- Group apt install/remove calls
- Handle fpm pool config by php version

(please, save the polar bears, squash this pr)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,137 +30,151 @@ Using ansible galaxy requirements file:
...
-| `manala_php_cli_configs`         | [ ]                  | Array   | PHP cli additional configurations.                     |
-| `manala_php_fpm_pools_dir`       | /etc/php5/fpm/pool.d | String  | PHP fpm pools directory path.                          |
-| `manala_php_fpm_pools_template`  | fpm_pools/empty.j2   | String  | Default pool template.                                 |
-| `manala_php_fpm_pools_exclusive` | false                | Boolean | Whether to remove all other non-specified pool files   |
-| `manala_php_fpm_pools`           | [ {name: www.conf} ] | Array   | PHP fpm pools configurations.                          |
+| Name                                          | Default                     | Type             | Description                                             |
+| --------------------------------------------- | --------------------------- | ---------------- | ------------------------------------------------------- |
+| `manala_php_version`                          | ~                           | String|Int|Float | Php version                                             |
```
Comment (hyvs) [inline README.md]: Be aware that `|` in `String|Int|Float` breaks markdown preview

File: tasks/configs.yml [hunk 2]
```diff
@@ -3,26 +3,26 @@
...
 - name: configs > {{ __manala_php_sapi|capitalize }} - Exclusive - Find files
   # configs files coming with php packages are nothing but symlinks,
   # so that we can distinguich them from our own files
   command: "find -type f -printf '%P\n'"
   args:
-    chdir: "{{ manala_php_configs_dir_base }}/{{ __manala_php_sapi }}/conf.d"
+    chdir: "{{ manala_php_versions[manala_php_version|string]['config_dir'] }}/{{ __manala_php_sapi }}/conf.d"
   changed_when: false
-  register: __manala_php_configs_sapi_find_output
+  register: __manala_php_configs_global_exclusive_find_result
```
Comment (hyvs) [inline tasks/configs.yml]: Should be named `__manala_php_configs_exclusive_find_result` ?

File: tasks/configs.yml [hunk 3]
```diff
@@ -3,26 +3,26 @@
 - name: configs > {{ __manala_php_sapi|capitalize }} - Templates
   template:
     src:  "{{ item.template|default(manala_php_configs_template|ternary(manala_php_configs_template, 'configs/empty.j2')) }}"
-    dest: "{{ manala_php_configs_dir_base }}/{{ __manala_php_sapi }}/conf.d/{{ item.file }}"
+    dest: "{{ manala_php_versions[manala_php_version|string]['config_dir'] }}/{{ __manala_php_sapi }}/conf.d/{{ item.file }}"
   notify:
-    - php {{ __manala_php_sapi }} restart
-  with_items: "{{ manala_php_configs + vars['manala_php_' + __manala_php_sapi + '_configs'] }}"
+    - php fpm restart
```
Comment (hyvs) [inline tasks/configs.yml]: We should not restart `php fpm` when others sapi configs are modified.

File: tasks/configs.yml [hunk 4]
```diff
@@ -3,26 +3,26 @@
...
   when: manala_php_configs_exclusive
 
 - name: configs > {{ __manala_php_sapi|capitalize }} - Exclusive - Remove files
   file:
-    path:  "{{ manala_php_configs_dir_base }}/{{ __manala_php_sapi }}/conf.d/{{ item }}"
+    path:  "{{ manala_php_versions[manala_php_version|string]['config_dir'] }}/{{ __manala_php_sapi }}/conf.d/{{ item }}"
     state: absent
   when: manala_php_configs_exclusive and (item not in (manala_php_configs + manala_php_{{ __manala_php_sapi }}_configs)|map(attribute='file'))
-  with_items: "{{ __manala_php_configs_sapi_find_output.stdout_lines|default([]) }}"
+  with_items: "{{ __manala_php_configs_global_exclusive_find_result.stdout_lines|default([]) }}"
```
Comment (hyvs) [inline tasks/configs.yml]: Same here.

**Review by hyvs — CHANGES_REQUESTED** [review]
first reading ⭐️

File: tasks/configs.yml (hunk 3 above)
Comment (nervo) [reply tasks/configs.yml]: Interesting point. I've digged a little, and fpm is the *only* sapi having an attached service. Regarding the fact that we can not (or do we ?) select what to notify, i notify onnly fpm, and check is fpm has been selected in the handler itself.
Another approach could be just to notify an agnostic "php restart", and check fpm in the handler too. Would you prefer ?

File: tasks/configs.yml (hunk 2 above)
Comment (nervo) [reply tasks/configs.yml]: Wo, nice catch

File: README.md (hunk 1 above)
Comment (nervo) [reply README.md]: Fixed (comma instead of pipe)

File: tasks/configs.yml (hunk 4 above)
Comment (nervo) [reply tasks/configs.yml]: Same wo

**Comment by nervo** [comment]
Wow, what an amazing piece of ~~crap~~ code, @nervo !

**Comment by nervo** [comment]
Oh, thank you, i'm just doing my job !

**Comment by nervo** [comment]
AUTO MEEEEERGE !

**Comment by chalasr** [comment]
Really, really nice job @nervo (hope @hyvs doesn't shot me, he requested changes)

**Pull request merged**